src/pages/home/report/ReportActionItemParentAction.tsx

```diff
@@ -1,80 +1,103 @@
 import useStyleUtils from '@hooks/useStyleUtils';
 import useThemeStyles from '@hooks/useThemeStyles';
 import useWindowDimensions from '@hooks/useWindowDimensions';
-import * as ReportActionsUtils from '@libs/ReportActionsUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import onyxSubscribe from '@libs/onyxSubscribe';
+import * as ReportUtils from '@libs/ReportUtils';
 import * as Report from '@userActions/Report';
 import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
 import type * as OnyxTypes from '@src/types/onyx';
 import AnimatedEmptyStateBackground from './AnimatedEmptyStateBackground';
 import ReportActionItem from './ReportActionItem';
 
 type ReportActionItemParentActionOnyxProps = {
-    /** The report currently being looked at */
+    /** The current report is displayed */
     report: OnyxEntry<OnyxTypes.Report>;
-
-    /** The actions from the parent report */
-    parentReportActions: OnyxEntry<OnyxTypes.ReportActions>;
 };
 
 type ReportActionItemParentActionProps = ReportActionItemParentActionOnyxProps & {
     /** Flag to show, hide the thread divider line */
     shouldHideThreadDividerLine?: boolean;
 
-    /** Flag to display the new marker on top of the comment */
-    shouldDisplayNewMarker: boolean;
-
     /** Position index of the report parent action in the overall report FlatList view */
     index: number;
 
     /** The id of the report */
     // eslint-disable-next-line react/no-unused-prop-types
     reportID: string;
-
-    /** The id of the parent report */
-    // eslint-disable-next-line react/no-unused-prop-types
-    parentReportID: string;
 };
 
-function ReportActionItemParentAction({report, parentReportActions = {}, index = 0, shouldHideThreadDividerLine = false, shouldDisplayNewMarker}: ReportActionItemParentActionProps) {
+function ReportActionItemParentAction({report, index = 0, shouldHideThreadDividerLine = false}: ReportActionItemParentActionProps) {
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
     const {isSmallScreenWidth} = useWindowDimensions();
-    const parentReportAction = parentReportActions?.[`${report?.parentReportActionID ?? ''}`] ?? null;
+    const ancestorIDs = useRef(ReportUtils.getAllAncestorReportActionIDs(report));
+    const [allAncestors, setAllAncestors] = useState<ReportUtils.Ancestor[]>([]);
+
+    useEffect(() => {
+        const unsubscribeReports: Array<() => void> = [];
+        const unsubscribeReportActions: Array<() => void> = [];
+        ancestorIDs.current.reportIDs.forEach((ancestorReportID) => {
+            unsubscribeReports.push(
+                onyxSubscribe({
+                    key: `${ONYXKEYS.COLLECTION.REPORT}${ancestorReportID}`,
+                    callback: () => {
+                        setAllAncestors(ReportUtils.getAllAncestorReportActions(report, shouldHideThreadDividerLine));
+                    },
+                }),
```

Comment:
This does not follow our Onyx practices. Components should use withOnyx and not Onyx.connect

Reply:
Could possibly have been a performance optimization? If we can explore getting rid of it that would be 👍

Reply:
> Components should use withOnyx and not Onyx.connect

Or `useOnyx` is a good option too now.